flink-core/src/main/java/org/apache/flink/configuration/Configuration.java

```diff
@@ -661,14 +665,16 @@ public boolean containsKey(String key) {
      */
     @PublicEvolving
     public boolean contains(ConfigOption<?> configOption) {
+        final boolean canBePrefixMap = canBePrefixMap(configOption);
+
+        // first try the current key
         synchronized (this.confData) {
-            // first try the current key
-            if (this.confData.containsKey(configOption.key())) {
+            if (containsInternal(configOption.key(), canBePrefixMap)) {
```

Comment:
Not a massive fan of having that logic spread on 3 different places. Could we extract some generic lookup method and lambda that acts on that?